tutorials/HDF5/Subset-HDF5-hyperspectral.Rmd

```diff
@@ -0,0 +1,241 @@
+h5createGroup(nf, "SJER/Reflectance/Metadata/Spectral_Data")
+
+```
+
+## Adding group attributes
+One of the great things about HDF5 files is that they can contain _data_ and _attributes_ within the same group. 
+As explained within the <a href="{{ site.baseurl }}/intro-hdf5-r-series" target="_blank"> Introduction to Working with HDF5 Format in R series</a>, attributes are a type of metadata that are associated with an HDF5 group or dataset. There may be multiple attributes associated with each group and/or dataset. Attributes come with a name and an associated array of information. In this tutorial, we will read the existing attribute data from the full hyperspectral tile using the `h5readAttributes()` function (which returns a `list` of attributes), then we loop through those attributes and write each attribute to its appropriate group using the `h5writeAttribute()` function.
+
+First, we will do this for the low-level "SJER/Reflectance" group. 
+
+```{r ref-attributes}
+
+a=h5readAttributes(f,"/SJER/Reflectance/")
+fid <- H5Fopen(nf)
+g=H5Gopen(fid, "SJER/Reflectance")
+
+for(i in 1:length(names(a))){
+h5writeAttribute(attr = a[[i]], h5obj=g, name=names(a[i]))
+}
+h5closeAll()
+```
+
+Next, we will loop through each of the datasets within the Coordinate_System group, and copy those (and their attributes, if present) from the full tile to our subset file. The Coordinate_System group contains many important pieces of information for geolocating our data, so we need to make sure that the sunset file has those data.
+
+```{r populate-group-attributes}
+
+# make a list of all groups within the full tile file
+ls=h5ls(f,all=T)
+
+#make a list of all of the names within the Coordinate_System group
+cg=unique(ls[ls$group=="/SJER/Reflectance/Metadata/Coordinate_System",]$name)
+
+for(i in 1:length(cg)){
+  print(cg[i])
+  a=h5readAttributes(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  d=h5read(f,paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]))
+  attributes(d)=a
+  h5write(obj=d,file=nf,
+          name=paste0("/SJER/Reflectance/Metadata/Coordinate_System/",cg[i]),
+          write.attributes=T)
+}
+```
+
+## Spectral Subsetting
+
+The goal of subsetting this dataset is to substantially reduce the file size to make it faster to download and process these data. While each AOP mosaic tile is not particularly large in terms of its spatial scale (1km by 1km at 1m resolution= 1,000,000 pixels, or about half as many pixels at shown on a standard 1080p computer screen), the 426 spectral bands available result in a fairly large file size. While having hundreds of bands is important for advanced research, we can use far fewer bands for demonstartion and teaching purposes, while still getting a good sense of what these hyperspectral data can do. Therefore, we will reduce the spectral resolution of these data by selecting every fourth band in the dataset, which reduces the file size to 1/4 of the original!
```

Comment:
1) add line breaks. 
2) typo "demonstartion"
3) if it make sense, consider adding a sentence or paragraph on occasions when spectral subsetting woudn't make sense.

Reply:
1. See above re: line breaks.
2. Typo corrected - good eye!
3. Not sure this is the most eloquent way to put it, but I added the information that you suggested and split this into two smaller paragraphs:
"The goal of subsetting this dataset is to substantially reduce the file size, making it faster to download and process these data. While each AOP mosaic tile is not particularly large in terms of its spatial scale (1km by 1km at 1m resolution= 1,000,000 pixels, or about half as many pixels at shown on a standard 1080p computer screen), the 426 spectral bands available result in a fairly large file size. Therefore, we will reduce the spectral resolution of these data by selecting every fourth band in the dataset, which reduces the file size to 1/4 of the original! 

Some circumstances demand the full spectral resolution file. For example, if you wanted to discern between the spectral signatures of similar minerals, or if you were conducting an analysis of the differences in the 'red edge' between plant functional types, you would want to use the full spectral resolution of the original hyperspectral dataset. Still, we can use far fewer bands for demonstration and teaching purposes, while still getting a good sense of what these hyperspectral data can do."